### PR TITLE
LPD-20282

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/LPD-20282

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:report:2561527465

**Comment by CrescenzoRegaLF** [comment]
ci:report:2561824531

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+	public static final String ACCOUNT_ENTRY_ID_VALIDATION_RULE_ENGINE =
+		"commerce-return-account-entry-id-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl.java:43]: isn't it better to start with a "false" statement and change it to true only at line 73 where it's the only case where we have a success?

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnCommerceOrderIdObjectValidationRuleEngineImpl.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+	public static final String COMMERCE_ORDER_ID_VALIDATION_RULE_ENGINE =
+		"commerce-return-commerce-order-id-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnCommerceOrderIdObjectValidationRuleEngineImpl.java:39]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnCommerceOrderStatusObjectValidationRuleEngineImpl.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+	public static final String COMMERCE_ORDER_STATUS_VALIDATION_RULE_ENGINE =
+		"commerce-return-commerce-order-status-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnCommerceOrderStatusObjectValidationRuleEngineImpl.java:44]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemAccountEntryIdObjectValidationRuleEngineImpl.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+	public static final String ACCOUNT_ENTRY_ID_VALIDATION_RULE_ENGINE =
+		"commerce-return-item-account-entry-id-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemAccountEntryIdObjectValidationRuleEngineImpl.java:44]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemCommerceOrderIdObjectValidationRuleEngineImpl.java [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+	public static final String COMMERCE_ORDER_ID_VALIDATION_RULE_ENGINE =
+		"commerce-return-item-commerce-order-item-id-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemCommerceOrderIdObjectValidationRuleEngineImpl.java:50]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemQuantityObjectValidationRuleEngineImpl.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+	public static final String QUANTITY_VALIDATION_RULE_ENGINE =
+		"commerce-return-item-quantity-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnItemQuantityObjectValidationRuleEngineImpl.java:46]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+	public static final String RETURN_STATUS_VALIDATION_RULE_ENGINE =
+		"commerce-return-return-status-validation-rule-engine";
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java:37]: same

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+import java.util.Map;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Crescenzo Rega
+ */
+@Component(service = ObjectValidationRuleEngine.class)
+public class CommerceReturnReturnStatusObjectValidationRuleEngineImpl
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java:26]: the name of this class sounds strange at least

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnableItemsObjectValidationRuleEngineImpl.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+	@Override
+	public Map<String, Object> execute(
+		Map<String, Object> inputObjects, String script) {
+
+		Map<String, Object> results = HashMapBuilder.<String, Object>put(
+			"validationCriteriaMet", true
+		).build();
+
+		// TODO This check must be done on the returned quantities
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnableItemsObjectValidationRuleEngineImpl.java]: remove the class if we are not ready to implement it
add it again when we have all we need

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl.java (hunk 1 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnAccountEntryIdObjectValidationRuleEngineImpl.java:43]: Hey @luca-pellizzon I followed the same approach as other validators already present

- [UniqueCompositeKeyObjectValidationRuleEngineImpl](https://github.com/CrescenzoRegaLF/liferay-portal/blob/LPD-20282/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/UniqueCompositeKeyObjectValidationRuleEngineImpl.java)

- [DDMObjectValidationRuleEngineImpl](https://github.com/CrescenzoRegaLF/liferay-portal/blob/LPD-20282/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/validation/rule/DDMObjectValidationRuleEngineImpl.java)

File: modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java (hunk 8 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/object/validation/rule/CommerceReturnReturnStatusObjectValidationRuleEngineImpl.java:26]: Hey @luca-pellizzon the approach was to use the first part of the name is the object name the second part is the field name, for this validator so we have the 'CommerceReturn' object and the 'ReturnStatus' field.
Let me know if we need to change the naming convention to something more readable

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:forward

**Pull request closed without merging**